rust/src/lib/policy/capture.rs

```diff
@@ -32,21 +32,72 @@ impl<'de> Deserialize<'de> for NetworkCaptureRules {
         let map = serde_json::Map::<String, serde_json::Value>::deserialize(
             deserializer,
         )?;
-        let mut cmds: Vec<(String, NetworkCaptureCommand)> = Vec::new();
 
+        // parse commands
+        let mut caps_map: HashMap<String, NetworkCaptureCommand> =
+            HashMap::new();
         for (k, v) in map.iter() {
             if let serde_json::Value::String(s) = v {
-                cmds.push((
+                caps_map.insert(
                     k.to_string(),
                     NetworkCaptureCommand::parse(s.as_str())
                         .map_err(serde::de::Error::custom)?,
-                ));
+                );
             } else {
                 return Err(serde::de::Error::custom(format!(
                     "Expecting a string, but got {v}"
                 )));
             }
         }
+
+        // Order cmds vector to honor how they get referenced, also
+        // detect looks and fail.
+        let mut visited_caps: HashSet<&String> = HashSet::new();
+        let mut cmds: Vec<(String, NetworkCaptureCommand)> = Vec::new();
+        for (cap_name, cap_cmd) in caps_map.iter() {
+            let mut cmd = cap_cmd;
+            // Implemented this with a stack so we don't have to do recursion
+            let mut stack: Vec<(&String, &NetworkCaptureCommand)> = Vec::new();
+            let mut stacked_caps: HashSet<&String> = HashSet::new();
+            while let Some(key_capture) = cmd.key_capture.as_ref() {
```

Comment:
I am scared of might lead to infinity loop for some YAML in the future due to poor patch.

Can you change the way to use method like `set_ifaces_up_priority()` by given a maximum recurrence count?

Reply:
> I am scared of this might lead to infinity loop for some YAML in the future due to poor patch.
> 

The PR include some testing, but we can increase the testing with more cases to ensure that we are good enough.

> Can you change the way to use method like `set_ifaces_up_priority()` by given a maximum recurrence count?

This means re-implement the solution using a counter ?

Reply:
Yes. Counter based. Like ordering controller/port, each round we set a `capture-priority` to its dependent's priority +1. And break the loop if all set. By having a limitation on loop count, we are preventing the dead loop at design instead of trusting test or developer covered all cases.

Reply:
This is different to controller ports, a port has just one owner, but a capture entry can be referenced by mutiple of them.

Reply:
Refactored to use a for loop with a limit and simplify it.